src/server/server_family.cc

```diff
@@ -114,12 +118,73 @@ void SubstituteFilenameTsPlaceholder(fs::path* filename, std::string_view replac
   *filename = absl::StrReplaceAll(filename->string(), {{"{timestamp}", replacement}});
 }
 
-string InferLoadFile(fs::path data_dir) {
+bool IsCloudPath(string_view path) {
+  return absl::StartsWith(path, "s3://");
```

Comment:
How about we define a global `constexpr auto kAwsPrefix = "s3://"sv;` and use it in all places?